app/gui/view/src/project_list.rs

```diff
@@ -274,11 +293,35 @@ impl ProjectList {
             grid_x <- grid_width.map(|width| -width / 2.0);
             grid_y <- all_with3(&content_size, &bar_height, &paddings, |s,h,p| s.y / 2.0 - *h - *p);
             _eval <- all_with(&grid_x, &grid_y, f!((x, y) grid.set_xy(Vector2(*x, *y))));
+
+            grid.reset_entries <+ frp.input.set_projects.map(|projects| {
+                let cols = 1;
+                let rows = projects.len();
+                (rows, cols)
+            });
+            grid.model_for_entry <+ grid.model_for_entry_needed.map2(
+                &frp.input.set_projects,
+                |(row, col), projects| {
+                    let (project_name, _) = projects.get(*row)?;
+                    Some((*row, *col, project_name.clone_ref()))
+                }
+            ).filter_map(|entry| entry.clone());
+
+            frp.source.selected_project <+ grid.entry_selected.map2(
+                &frp.input.set_projects,
+                |selected_entry, projects| {
+                    let (row, _) = (*selected_entry)?;
+                    projects.get(row).cloned()
+                }
+            );
+            some_project_selected <- frp.output.selected_project
+                .filter_map(|selection| selection.clone());
+            grid.select_entry <+ some_project_selected.constant(None);
```

Comment:
these FRP definitions are way more complex than any other FRP definition above. Could you try refactoring them to keep them simpler (shorter in the amount of lines per FRP definition), just as it's done in other places, please? Long FRP definitions require reader to spend a lot of time understanding every FRP binding, while the FRP code should focus on the interaction logic instead. Refactoring things to helper functions is often the best idea in such places.

Reply:
Hmmm, I'm not sure in what sense you want them simplified. Could you give an example?

Most of the code here was moved here from `presenter::project`, but I can have a look at refactoring it if you can give some guidance.

Reply:
I'll give a few examples here, but you need to go trough available FRP functions and see what can be applied here to simplify these definitions.

For example:

```
some_project_selected <- frp.output.selected_project
     .filter_map(|selection| selection.clone());
```

Is the same as

```
some_project_selected <= frp.output.selected_project;
```

Things such as this:

```
grid.reset_entries <+ frp.input.set_projects.map(|projects| {
                 let cols = 1;
                 let rows = projects.len();
                 (rows, cols)
             });
```

can be defined as:

```
// We are using `p` here, as it is obvious that it refers to `projects` as it is defined just 
// a few chars after the reference to `input.set_projects`. Ofc this comment is for you,
// should not land in the real code ;)
rows_and_cols <- frp.input.set_projects.map(|p| (p.len(), 1)); 
grid.reset_entries <+ rows_and_cols;
```